nodl_python/nodl/_parsing/_v1/_parsing.py

```diff
@@ -15,21 +15,25 @@
 from lxml import etree
 from nodl import errors
 from nodl._parsing._schemas import v1_schema
-from nodl._util import get_bool_attribute
-from nodl.types import Action, Node, Parameter, Service, Topic
+from nodl.types import (
+    Action,
+    Node,
+    Parameter,
+    PubSubRole,
+    ServerClientRole,
+    Service,
+    Topic,
+)
 
 
 def _parse_action(element: etree._Element) -> Action:
     """Parse a NoDL action from an xml element."""
     name = element.get('name')
     action_type = element.get('type')
 
-    server = get_bool_attribute(element, 'server')
-    client = get_bool_attribute(element, 'client')
-    if not (server or client):
-        raise errors.AmbiguousActionInterfaceError(element)
+    role = ServerClientRole(element.get('role'))
```

Comment:
This could raise a `ValueError`. Are you just relying on the schema to catch it?